### ISSUE TITLE
Health check for WITH DOCKER

### ISSUE THREAD

**Issue by agbell** [earthly/earthly#656]
Docker images support a HEALTHCHECK statement. WITH DOCKER could check health of started images for images that support it.  

in dockerfile:
```
HEALTHCHECK CMD wget -q --method=HEAD localhost/system-status.txt
```
Check like this:
 ```
docker inspect --format "{{json .State.Health }}" <<image_name>>

```
Not many images seem to have a HEALTHCHECK set, so this may be a low priority but github actions uses this for services you ask it to start.

**Comment by vladaionescu** [comment]
This makes sense. I wish docker-compose had native support for this stuff. Then it could enforce startup order with the right waiting included. In any case, I think an addition in Earthly would be pretty useful.

This would only work for `--compose` in the beginning.

If you use `docker run -d` or `docker-compose up -d` manually you'd have to do your own waiting logic commands. Maybe when we support more than one `RUN` command within `WITH DOCKER`, we could add a `WAIT` command so you can insert this kind of waiting anywhere.

**Comment by frafra** [comment]
> I wish docker-compose had native support for this stuff.

```
services:
  app:
    depends_on:
      database:
        condition: service_healthy
```

**Comment by xsteadfastx** [comment]
are there any news on this? i have a healthcheck definded in my docker-compose but the docker-compose in earthly doesnt wait till its healthy.

**Comment by andyli** [comment]
Maybe Earthly should just use `--wait` (or let users opt-in) when starting docker-compose?

https://docs.docker.com/engine/reference/commandline/compose_up/#options

**Comment by xsteadfastx** [comment]
yes thats exactly what im missing.

**Comment by gnuletik** [comment]
Yes it could be really nice to add a `--compose-wait` option to the `WITH DOCKER` command which would insert the `--wait` argument to `docker compose up`.
